crates/rsonpath-lib/src/classification/structural/nosimd.rs

```diff
@@ -54,9 +54,11 @@ impl<'a> Iterator for Block<'a> {
             match character {
```

Comment:
What about this instead:
```suggestion
            let block = match character {
                _ if is_quoted => (),
                b':' if self.are_colons_on => Some(Colon(self.idx1)),
                b'{' => Some(OpeningBrace(self.idx)),
                b'[' => Some(OpeningBracket(self.idx)),
                b',' if self.are_commas_on => Some(Comma(self.idx)),
                b'}' => Some(ClosingBrace(self.idx)),
                b']' => Some(ClosingBracket(self.idx)),
                _ => (),
            };

            self.idx += 1;

            (block.is_some()).then(|| {
                return block;
            });
```

Reply:
I applied some variation on this, but used `if` at the end instead of `bool::then`.